pkg/checks/checks.go

```diff
@@ -30,23 +25,9 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}
 	return wait.PollUntil(time.Second, func() (bool, error) {
```

Comment:
Can we increase PollUntil time?  5 second might be too much, 2? 
```
+ go run cmd/sync/sync.go -run-once=true
2018/08/06 08:58:49 Sync process started!
Get https://mjudeikis-test.eastus.cloudapp.azure.com/healthz: dial tcp 255.255.255.255:443: connect: network is unreachable
.... x ~50 times
Get https://mjudeikis-test.eastus.cloudapp.azure.com/healthz: dial tcp 255.255.255.255:443: connect: network is unreachable

Get https://mjudeikis-test.eastus.cloudapp.azure.com/healthz: EOF
... x ~20 times
Get https://mjudeikis-test.eastus.cloudapp.azure.com/healthz: EOF
Get https://mjudeikis-test.eastus.cloudapp.azure.com/healthz: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
2018/08/06 09:02:28 Update Namespace/default
2018/08/06 09:02:28 - Object.map[metadata].map[annotations]: <does not have key> != map[openshift.io/node-selector:]
2018/08/06 09:02:28 - Object.map[spec].map[finalizers].slice[1]: <no value> != openshift.io/origin
2018/08/06 09:02:28 Skip Namespace/kube-public

```

Reply:
I am seeing the same set of errors, only for longer than just 2 or 5 seconds. We need to ignore these errors inside the loop, increasing the interval is not a robust solution.

Reply:
They can't be ignored forever. There is a change to failure here. In example, if DNS was allocated for us, but we can't claim it from backend due ARM issues.  I think I need to set a deadline for this particular piece of code. Something like 5 min should be more than enough for infrastructure to converge?

Reply:
A timeout is acceptable I think.

Reply:
``` 
ctxTimeout, cancel := context.WithTimeout(ctx, time.Minute*5)
defer cancel()
	return wait.PollUntil(time.Second, func() (bool, error) {
		resp, err := cli.Do(req)
		if err != nil {
			fmt.Println(err)
			return false, nil
		}
		return resp != nil && resp.StatusCode == http.StatusOK, nil
	}, ctxTimeout.Done())
``` 
?